### PR TITLE
US118255 Include 'codeMinor' from outcome failure responses in thrown errors to help with error handling

### PR DESCRIPTION
**Pull request by MarquisLP**
Note that all of the Javascript files had to be re-generated for testing. As such, the only files relevant for review are the `.coffee` files.

### CODE REVIEW

File: lib/redis-nonce-store.js [hunk 1]
```diff
@@ -49,15 +108,20 @@
...
       this.redis.expire(nonce, EXPIRE_IN_SEC);
+      _$jscoverage["redis-nonce-store.coffee"][42]++;
       return next(null);
     };
 
     return RedisNonceStore;
 
   })(NonceStore);
 
+  _$jscoverage["redis-nonce-store.coffee"][45]++;
```
Comment (rschick) [inline lib/redis-nonce-store.js:123]: Did something happen to add all this coverage instrumentation to the compiled code?

File: src/errors.coffee [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
-  constructor: (@message) ->
-    super
+  constructor: (message) ->
+    super(message)
 class OutcomeResponseError extends Error
-  constructor: (@message) ->
-    super
+  constructor: (message, @codeMinor) ->
+    super(message)
+    this.message = message
```
Comment (rschick) [inline src/errors.coffee:22]: In CoffeeScript, would this be ```
super message
@message = message`
```

File: src/errors.coffee [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
 class SignatureError extends Error
-  constructor: (@message) ->
-    super
+  constructor: (message) ->
+    super(message)
 class NonceError extends Error
-  constructor: (@message) ->
-    super
+  constructor: (message) ->
+    super(message)
```
Comment (rschick) [inline src/errors.coffee:18]: Why did you change all these?

File: lib/redis-nonce-store.js (hunk 1 above)
Comment (MarquisLP) [reply lib/redis-nonce-store.js:123]: I'm not sure... I tried re-generating with the older CoffeeScript version (1.10.0) used in the newest master commit, and it still added these coverage lines to the files.

File: src/extensions/outcomes.coffee [hunk 4]
```diff
@@ -135,10 +135,12 @@ class OutcomeService
     @_send_request doc, (err, result, xml) =>
       return callback(err, result) if err
 
-      score = parseFloat navigateXml(xml, 'imsx_POXBody.readResultResponse.result.resultScore.textString'), 10
+      score = navigateXml(xml, 'imsx_POXBody.readResultResponse.result.resultScore.textString')
```
Comment (rschick) [inline src/extensions/outcomes.coffee:138]: Did you add/modify tests to cover these changes?

File: src/extensions/outcomes.coffee [hunk 5]
```diff
@@ -203,14 +205,15 @@ class OutcomeService
 
   _process_response: (body, callback) ->
     xml2js.parseString body, trim: true, (err, result) =>
-      return callback new errors.OutcomeResponseError('The server responsed with an invalid XML document'), false if err
+      return callback new errors.OutcomeResponseError('The server responsed with an invalid XML document', 'invaliddata'), false if err
```
Comment (rschick) [inline src/extensions/outcomes.coffee:208]: Same, is this covered in tests?

File: src/errors.coffee (hunk 2 above)
Comment (MarquisLP) [reply src/errors.coffee:22]: Yeah, that works and looks more in style with CoffeeScript, so I'll make the change.

**Comment by rschick** [comment]
The `.js` is not only for testing. It is what is going to be deployed, since we are installing this module directly from github as opposed to from an artifact repo like artifactory or npm.

File: src/errors.coffee (hunk 3 above)
Comment (MarquisLP) [reply src/errors.coffee:18]: [Starting from CoffeeScript 2, CoffeeScript no longer allows "bare" super calls.](http://coffeescript.org/announcing-coffeescript-2/)

File: src/extensions/outcomes.coffee (hunk 4 above)
Comment (MarquisLP) [reply src/extensions/outcomes.coffee:138]: Nope, good catch. I'll add some for this and the one below.

File: lib/redis-nonce-store.js (hunk 1 above)
Comment (MarquisLP) [reply lib/redis-nonce-store.js:123]: `make build` worked as you said in removing all the coverage code. Will commit this as well.

File: src/extensions/outcomes.coffee (hunk 4 above)
Comment (MarquisLP) [reply src/extensions/outcomes.coffee:138]: Actually, this change is covered by the tests in the `describe('read')` block in the OutcomesExtension test file.

File: src/extensions/outcomes.coffee (hunk 5 above)
Comment (MarquisLP) [reply src/extensions/outcomes.coffee:208]: This one needed tests even before the changes, so I'll look into adding some now.

File: src/errors.coffee [hunk 6]
```diff
@@ -1,24 +1,25 @@
...
 class ParameterError extends Error
-  constructor: (@message) ->
-    super
+  constructor: (message) ->
+    super(message)
 class SignatureError extends Error
-  constructor: (@message) ->
-    super
+  constructor: (message) ->
+    super(message)
```
Comment (rschick) [inline src/errors.coffee:15]: Coffeescript style is just `super message` without parens.
